packages/framework/data-object-base/src/runtimeFactory.ts

```diff
@@ -21,6 +23,9 @@ export class RuntimeFactory extends RuntimeFactoryHelper {
 		private readonly defaultStoreFactory: IFluidDataStoreFactory,
 		storeFactories: IFluidDataStoreFactory[] = [defaultStoreFactory],
 		private readonly requestHandlers: RuntimeRequestHandler[] = [],
+		private readonly initializeEntryPoint?: (
```

Comment:
i'm trying to see how the switch from request handler to entrypoint would work here. i think we might want to fork this class, as i don't think its possible to move to entrypoint, and then not have to worry about requestHandlers, as given the constructor the consumer will still get a breaking change

Reply:
That applies to `BaseContainerRuntimeFactory` and `ContainerRuntimeFactoryWithDefaultDataStore` as well, right? Forking all of them makes me a bit concerned with the names we would end up with for new versions, unless maybe we consolidate them into a single new one that takes a property bag as its only constructor parameter, like we did with `ContainerRuntime.load() -> ContainerRuntime.loadRuntime()`.

Do we have a story about the role of each of the three factories? I couldn't find any uses of `RuntimeFactory` in our repo nor our partner repos; in the Bohemia codebase I did find an extension of our `BaseContainerRuntimeFactory`, besides their own `BaseContainerRuntimeFactory<T>`, and in general I found uses of our `ContainerRuntimeFactoryWithDefaultDataStore`.

`BaseContainerRuntimeFactory` plus its extension `ContainerRuntimeFactoryWithDefaultDataStore` seem to be the more complete version, taking in `IFluidDependencySynthesizer` and `IContainerRuntimeOptions`, and passing `context.scope` to the container runtime, which `RuntimeFactory` does not (although the full `context` is passed in both cases). `RuntimeFactory` also does not expose its data store registry (which is typed differently than in the other runtime factory classes), but not sure how relevant it is for it to be exposed in the factory itself. Finally, `BaseContainerRuntimeFactory` has the `containerInitializingFirstTime` and `containerHasInitialized` extension points not present in `RuntimeFactory`.

All that makes me lean towards forking, if anything, `BaseContainerRuntimeFactory` and `ContainerRuntimeFactoryWithDefaultDataStore`, and fully deprecating `RuntimeFactory` in favor of the other two, but probably do that as follow-up work when we actually remove the request pattern.

@anthony-murphy thoughts?

Tangent: just noticed that `ContainerRuntimeFactoryWithDefaultDataStore` also needs `initializeEntryPoint` support, adding that now.

Reply:
Or maybe `ContainerRuntimeFactoryWithDefaultDataStore` also goes away, and consumers should do that themselves in the entryPoint initialization function.

Reply:
IMO `ContainerRuntimeFactoryWithDefaultDataStore` should be considered "not recommended" at this point (especially in a post-entrypoint world where the entrypoint takes on most of the duties of the default data store).  It doesn't do enough simplification to justify the extra baggage it brings along.

Honestly it would probably be best to look at how we can be comfortable ditching the base class runtime factories (all these plus maybe even `RuntimeFactoryHelper`.  Creating an `IRuntimeFactory` from scratch is not super hard (especially if you're not exercising the request handler pattern), and even the slightly annoying bits could maybe be solved rather than cover them up with a base class.

Reply:
I tend to agree, for the post-entryPoint world. It's unfortunate that's the one with most usage (as far as I could see) in our consumers' repos.